src/clojure/nrepl/middleware/interruptible_eval.clj

```diff
@@ -121,8 +121,10 @@
                                                      :ns (-> *ns* ns-name str)})))
            ;; TODO: customizable exception prints
            :caught (fn [e]
-                     (let [root-ex (#'clojure.main/root-cause e)]
-                       (when-not (instance? ThreadDeath root-ex)
+                     (let [root-ex (#'clojure.main/root-cause e)
```

Comment:
For the sake of future readers of the code it'd be nice if you added here the explanations for why this is needed, as it's not immediately obvious.